ingress/controllers/nginx/controller.go

```diff
@@ -217,6 +257,75 @@ func (lbc *loadBalancerController) getUDPConfigMap(ns, name string) (*api.Config
 	return lbc.client.ConfigMaps(ns).Get(name)
 }
 
+func (lbc *loadBalancerController) updateEpNamedPorts(key string) {
+	if !lbc.controllersInSync() {
+		lbc.svcEpQueue.requeue(key, fmt.Errorf("deferring sync till endpoints controller has synced"))
+		return
+	}
+
+	svcObj, svcExists, err := lbc.svcLister.GetByKey(key)
+	if err != nil {
+		glog.Warningf("error getting service %v: %v", key, err)
+		return
+	}
+
+	if !svcExists {
+		glog.Warningf("service %v not found", key)
+		return
+	}
+
+	svc := svcObj.(*api.Service)
+	if svc.Spec.Selector == nil {
+		return
+	}
+
+	pods, err := lbc.client.Pods(svc.Namespace).List(api.ListOptions{
```

Comment:
I'd suggest you don't actually list all pods. This is a really heavy operation which if done too often can have all sorts of side effects (oom on this container in large clusters -- this happened with kube2sky in 1000 nodes, load on apiserver, request throttling etc). The only sane way to retrieve all pods from apiserver (with or without a label selector) is via watch.

I'm fine with a todo for now. 
I dislike the growing dependency graph. 

Can you do this less often? (currently it happens every svc enqueue right?)